d4s2_api_v2/serializers.py

```diff
@@ -54,3 +55,71 @@ class Meta:
     @staticmethod
     def get_duke_ds_user(user):
         return DDSUtil(user).get_current_user().id
+
+
+class S3EndpointSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = S3Endpoint
+        resource_name = 's3endpoints'
+        fields = ('id', 'url',)
+
+
+class S3UserSerializer(serializers.ModelSerializer):
+    email = serializers.SerializerMethodField()
+    type = serializers.SerializerMethodField()
+
+    class Meta:
+        model = S3User
+        resource_name = 's3users'
+        fields = ('id', 'user', 'endpoint', 'email', 'type')
+
+    @staticmethod
+    def get_email(s3_user):
```

Comment:
I think this behavior can be accomplished without extra methods by using the [source](http://www.django-rest-framework.org/api-guide/fields/#source) keyword:

```
email = serializers.CharField(source='user.email')
type = serializers.CharField(source='get_type_label')
```